python/cudf/cudf/pylibcudf_tests/test_binaryops.py

```diff
@@ -0,0 +1,247 @@
+    [
+        ("int64", "int64", "int64"),
+        ("int64", "float64", "float64"),
+        ("int64", "int64", "datetime64[ns]"),
+    ],
+    indirect=True,
+    ids=idfn,
+)
+def test_add(pa_data, plc_data):
+    def add(x, y):
+        if x is None or y is None:
+            return None
+        return x + y
+
+    _test_binaryop_inner(
+        pa_data,
+        plc_data,
+        add,
+        plc.binaryop.BinaryOperator.ADD,
+    )
+
+
+@pytest.mark.parametrize(
+    "pa_data",
+    [("int64", "int64", "int64"), ("int64", "float64", "float64")],
+    indirect=True,
+    ids=idfn,
+)
+def test_sub(pa_data, plc_data):
+    def sub(x, y):
+        if x is None or y is None:
+            return None
+        return x - y
+
+    _test_binaryop_inner(
+        pa_data,
+        plc_data,
+        sub,
+        plc.binaryop.BinaryOperator.SUB,
+    )
+
+
+@pytest.mark.parametrize(
+    "pa_data",
+    [("int64", "int64", "int64"), ("int64", "float64", "float64")],
+    indirect=True,
+    ids=idfn,
+)
+def test_mul(pa_data, plc_data):
+    def mul(x, y):
+        if x is None or y is None:
+            return None
+        return x * y
+
+    _test_binaryop_inner(
+        pa_data,
+        plc_data,
+        mul,
+        plc.binaryop.BinaryOperator.MUL,
+    )
+
+
+@pytest.mark.parametrize(
+    "pa_data",
+    [("int64", "int64", "int64"), ("int64", "float64", "float64")],
+    indirect=True,
+    ids=idfn,
+)
+def test_div(pa_data, plc_data):
+    def div(x, y):
+        if x is None or y is None:
+            return None
+        return x / y
+
+    _test_binaryop_inner(
+        pa_data,
+        plc_data,
+        div,
+        plc.binaryop.BinaryOperator.DIV,
+    )
+
+
+@pytest.mark.parametrize(
+    "pa_data",
+    [("int64", "int64", "int64"), ("int64", "float64", "float64")],
+    indirect=True,
+    ids=idfn,
+)
+def test_floordiv(pa_data, plc_data):
+    def floordiv(x, y):
+        if x is None or y is None:
+            return None
+        return x // y
+
+    _test_binaryop_inner(
+        pa_data,
+        plc_data,
+        floordiv,
+        plc.binaryop.BinaryOperator.FLOOR_DIV,
+    )
+
+
+@pytest.mark.parametrize(
+    "pa_data",
+    [("int64", "int64", "int64"), ("int64", "float64", "float64")],
+    indirect=True,
+    ids=idfn,
+)
+def test_truediv(pa_data, plc_data):
+    def truediv(x, y):
+        if x is None or y is None:
+            return None
+        return x / y
+
+    _test_binaryop_inner(
+        pa_data,
+        plc_data,
+        truediv,
+        plc.binaryop.BinaryOperator.TRUE_DIV,
+    )
+
+
+@pytest.mark.parametrize(
+    "pa_data",
+    [("int64", "int64", "int64"), ("int64", "float64", "float64")],
+    indirect=True,
+    ids=idfn,
+)
+def test_mod(pa_data, plc_data):
+    def mod(x, y):
+        if x is None or y is None:
+            return None
+        return x % y
+
+    _test_binaryop_inner(
+        pa_data,
+        plc_data,
+        mod,
+        plc.binaryop.BinaryOperator.MOD,
```

Comment:
Python's modulus operator is implemented as PYMOD. We probably need more expansive inputs to test this code accurately.
```suggestion
        plc.binaryop.BinaryOperator.PYMOD,
```

Reference: https://stackoverflow.com/questions/1907565/c-and-python-different-behaviour-of-the-modulo-operation

Reply:
> We probably need more expansive inputs to test this code accurately.

The goal of the pylibcudf tests is generally to test that the wrappers are implemented correctly: we defer to libcudf for covering the implementation correctness of all the kernels.

So I guess in this case we would want an input that can differentiate between getting `MOD` and `PYMOD` mixed up in the wrapper.